zebra-network/src/peer_set/set.rs

```diff
@@ -995,78 +1004,74 @@ where
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        self.poll_background_errors(cx)?;
-
-        // Update peer statuses
+        // Update service and peer statuses.
+        //
+        // # Correctness
+        //
+        // If we poll these futures in series, then only the last polled future will wake
```

Comment:
This is wrong:
https://github.com/ZcashFoundation/zebra/pull/7859/files#r1380722199